File: edental_flutter_main/lib/models/payment.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';

import '../helpers/dateJsonFormatter.dart';

@jsonSerializable
class Payment {
  int? id;
  @JsonProperty(converter: CustomDateTimeConverter())
  DateTime date;
  double amount;
  int userId;
  int treatmentId;
  int appointmentId;
  String? client;
  String? treatmentName;
  String? dentistName;
  String? paymentIntent;
  Payment(
      this.date, this.amount, this.userId, this.treatmentId, this.appointmentId,
      {this.client,
      this.treatmentName,
      this.id,
      this.dentistName,
      this.paymentIntent});
}

File: DentalOffice.Entities/dbEntities/Payment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DentalOffice.Entities.dbEntities
{
    public class Payment
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public decimal Amount { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int AppointmentId { get; set; }
        public Appointment Appointment { get; set; }
        public string PaymentIntent { get; set; }
        [NotMapped]
        public string DentistId { get; set; }
        [NotMapped]
        public string Client { get; set; }
        [NotMapped]
        public string TreatmentName { get; set; }
    }
}
